modules/search/wikileaks.py

```diff
@@ -0,0 +1,51 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+meta = {
+	'name': 'Wikileaks Search',
+	'author': 'Aman Singh',
+	'version': '0.1',
+	'description': 'WikiLeaks is an international non-profit organisation that publishes news leaks and classified media provided by anonymous sources.',
+	'sources': ('wikileaks',),
+	'options': (
+		('query', None, True, 'Query string', '-q', 'store', str),
+		('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store', int),
+	),
+	'examples': ('wikileaks -q <QUERY> -l 15 --output',)
+}
+
+def module_api(self):
+	query = self.options['query']
+	limit = self.options['limit']
+	run = self.wikileaks(query, limit)
+	run.run_crawl()
+
+	output = {'results': []}
+	links = run.links_with_data
+
+	for item in links:
```

Comment:
output['results'] = links isn't better?

Reply:
the output contains both links and titles of results 
```
self._links_with_data.append({
				'title': result.find_all('a')[0].text,
				'link': result.find_all('a')[0]['href']
				})
```

Reply:
the output format off links_with_data is not good. it should be something like this: [{'title': '....', 'link': '...}, ...] or [(title, link), (title, link), ...]
Also for the summer, we should organize/integrate the sources to have a consistent and similar result format.

Reply:
well the format is like  ```[{'title': '....', 'link': '...}, ...]``` only, links_with_data is an array.

Reply:
The result format can be something like ```[{'link': '...', 'key': 'value', ....}, {'link': '...', 'key': 'value', ....}, ......]```